handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -297,6 +297,11 @@ protected void deallocate() {
             int options = SSLContext.getOptions(ctx) |
                           SSL.SSL_OP_NO_SSLv2 |
                           SSL.SSL_OP_NO_SSLv3 |
+                          // Disable TLSv1 and TLSv1.1 by default as these are not considered secure anymore
+                          // and the JDK is doing the same:
+                          // https://www.oracle.com/java/technologies/javase/8u291-relnotes.html
+                          SSL.SSL_OP_NO_TLSv1 |
+                          SSL.SSL_OP_NO_TLSv1_1 |
```

Comment:
Bcz I don't have much context on how OpenSSL should be configured, have some questions:
1. Why do we have 
```
int protocolOpts = SSL.SSL_PROTOCOL_SSLV3 | SSL.SSL_PROTOCOL_TLSV1 |
                                   SSL.SSL_PROTOCOL_TLSV1_1 | SSL.SSL_PROTOCOL_TLSV1_2;
```
above? What it does?

2. TLSv1.1 may still be required in some use-cases. Is there any way users can enable it back? JDK still allows using it with some additional configuration.

3. Do we need a test to verify it's disabled by default but can be enabled if necessary?

Reply:
1) This basically set what protocol versions can be enabled / disabled on the fly.

2) See above... you can still enable it. It will just not be enabled by default.

3) Yep let me add something

Reply:
Thanks for clarification.

While we are here, one more question: should we disable SSLv3 completely bcz it's vulnerable? https://github.com/netty/netty/commit/a1af35313cc8414109e7a411bb2401ae31046289

Reply:
@idelpivnitskiy pushed tests

Reply:
@idelpivnitskiy yes... let me do this as an extra PR tho